eth/fetcher/tx_fetcher.go

```diff
@@ -261,10 +265,15 @@ func (f *TxFetcher) Notify(peer string, types []byte, sizes []uint32, hashes []c
 		default:
 			unknownHashes = append(unknownHashes, hash)
 			if types == nil {
-				unknownMetas = append(unknownMetas, nil)
+				unknownMetas = append(unknownMetas, txMetadata{arrival: f.counter})
 			} else {
-				unknownMetas = append(unknownMetas, &txMetadata{kind: types[i], size: sizes[i]})
+				if sizes[i] == 0 {
+					// invalid size parameter, return error
+					return fmt.Errorf("announcement from tx %x had an invalid 0 size metadata", hash)
+				}
```

Comment:
I don't think we should special case the size check here. The same way 0 is bad, so is 1 and 2 and probably a few more. We also don't really expect getting sent 0 unless it's a faulty remote client (i.e. a malicious can send 1). Also 0 will fail post-validation after we retrieve the hash, just as say 1 would. Cleaner just to not care about it at this point.

Reply:
removed.